### PR TITLE
Stapler: Beta process proposal

### PR DESCRIPTION
**Pull request by gblakeman**
Proposes a new way to handle `beta` builds (based on my previous feature-flipping work for “public” and “internal” build targets):

* Adds a new option to the script that generates the manifest file (`genmanifest:beta`). This tweaked script will generate `manifest.json` inside a local `beta-build` directory. The script expects to find a `.plugin-beta-id` file. It uses the existing plugin name and appends “ (Beta)”, so “_Stapler_” becomes “_Stapler (Beta)_” in the new manifest.
* Updates our webpack config to publish source into `beta-build/dist` instead of `dist` when the `beta` featureset flag is available. This works for both `production` and in-development `watch` builds:
  * `$ npm run build:beta`
  * `$ npm run watch:beta`
* Adds a new `beta` feature flag option that can be accessed in the runtime:
  * `if (process.env.FEATURESET === 'beta') { // do a thing }`

The end result of this setup will let us create beta builds inside our local project directories/repos without needing to maintain a separate repo, copy/re-install node modules, etc. It also gives us the possibility to restrict certain features to the beta builds separately from the production builds, even on the same branch.

In your local repo directory, point Figma at the root manifest for the production version (i.e. `stapler/manifest.json` in this repo) and then separately at `stapler/beta-build/manifest.json` for the beta build.

plugin-manifest-watch

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -9,5 +10,5 @@ npm-debug.log
 # config
 .env
 .plugin-id
+.plugin-beta-id
 .plugin-name
-manifest.json
```
Comment (gblakeman) [inline .gitignore:13]: Just moving this up to the `# builds` section since it is technically a build file.

File: stapler/package.json [hunk 2]
```diff
@@ -10,24 +10,27 @@
...
+    "watch:public": "npm run genmanifest && npx webpack --env featureset=public --mode=development --watch",
+    "watch:internal": "npm run genmanifest && npx webpack --env featureset=internal --mode=development --watch",
+    "watch:beta": "npm run genmanifest:beta && npx webpack --env featureset=beta --mode=development --watch",
     "start": "npm run watch",
     "start:public": "npm run watch:public",
     "start:internal": "npm run watch:internal",
+    "start:beta": "npm run watch:beta",
     "postinstall": "npm run build",
     "genmanifest": "cp manifest.example.json manifest.json && plugin_id=$(cat '.plugin-id') && sed -i.tmp s/PLUGIN_ID/$plugin_id/ manifest.json && plugin_name=$(cat '.plugin-name') && sed -i.tmp \"s/PLUGIN_NAME/$plugin_name/g\" manifest.json && rm manifest.json.tmp",
+    "genmanifest:beta": "rm -rf beta-build && mkdir beta-build && cp manifest.example.json beta-build/manifest.json && plugin_id=$(cat '.plugin-beta-id') && sed -i.tmp s/PLUGIN_ID/$plugin_id/ beta-build/manifest.json && plugin_name=$(cat '.plugin-name') && sed -i.tmp \"s/PLUGIN_NAME/$plugin_name (Beta)/g\" beta-build/manifest.json && rm beta-build/manifest.json.tmp",
```
Comment (joeyyang) [inline stapler/package.json:26]: would the `manifest.json` file itself have to change the directory of where it looks for the `main` file?

**Review by joeyyang — APPROVED** [review]
LGTM!

**Review by lynnerang — APPROVED** [review]
Looks great, thanks for simplifying the beta setup!

**Pull request merged**